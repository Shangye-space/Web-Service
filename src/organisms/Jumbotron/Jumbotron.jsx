import React from 'react';

import Intro from '../../atoms/Intro';

import classnames from 'classnames';
import styles from './Jumbotron.module.scss';

const Jumbotron = () => {
    return (
        <>
            <section
                className={classnames(
                    styles.wrapper,
                    'jumbotron',
                    'jumbotron-fluid',
                )}
            >
                <Intro />
                <div className="container h-100">
                    <div className="d-flex h-100 text-center align-items-center">
                        <div
                            className={classnames(
                                styles.textArea,
                                'w-100',
                                'text-white',
                            )}
                        >
                            <h1 className="display-3">
                                Shangye
                                <strong className={styles.orangeText}>
                                    .space
                                </strong>
                            </h1>
                            <p className="lead mb-0">
                                ALWAYS BE
                                <br />
                                YOURSELF
                            </p>
                        </div>
                    </div>
                </div>
            </section>
        </>
    );
};

export default Jumbotron;
