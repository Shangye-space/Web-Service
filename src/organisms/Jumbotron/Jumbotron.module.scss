@import '../../settings/variables.scss';

.wrapper {
    position: relative;
    background-color: black;
    height: 100vh;
    min-height: 25rem;
    width: 100%;
    overflow: hidden;

    video {
        position: absolute;
        top: 50%;
        left: 50%;
        min-width: 100%;
        min-height: 100%;
        width: auto;
        height: auto;
        z-index: 0;
        -ms-transform: translateX(-50%) translateY(-50%);
        -moz-transform: translateX(-50%) translateY(-50%);
        -webkit-transform: translateX(-50%) translateY(-50%);
        transform: translateX(-50%) translateY(-50%);
    }

    .container {
        position: relative;
        z-index: 2;
    }

    .overlay {
        position: absolute;
        top: 0;
        left: 0;
        height: 100%;
        width: 100%;
        background-color: $black;
        opacity: 0.5;
        z-index: 1;
    }

    @media (pointer: coarse) and (hover: none) {
        .wrapper {
            background: url('https://source.unsplash.com/XT5OInaElMw/1600x900')
                black no-repeat center center scroll;

            video {
                display: none;
            }
        }
    }
}

.textArea {
    z-index: 99;
    text-align: left;
    h1 {
        font-size: 84px;
        font-family: $marker;
    }

    p {
        font-size: 76px;
        font-family: $casual;
        background-color: rgba(0, 0, 0, 0.5);
        box-sizing: border-box;
        display: block;
        width: 559px;
        padding: 0 32px;
        line-height: 1.1;
    }
}

.orangeText {
    color: $soft-orange;
}
