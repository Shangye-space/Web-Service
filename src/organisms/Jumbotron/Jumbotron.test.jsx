import React from 'react';
import ReactDOM from 'react-dom';
import Jumbotron from './Jumbotron.jsx';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Jumbotron />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('snapshot renders', () => {
    const component = renderer.create(<Jumbotron />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
