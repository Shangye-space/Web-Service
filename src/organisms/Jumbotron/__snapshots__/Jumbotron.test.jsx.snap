// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<section
  className="wrapper jumbotron jumbotron-fluid"
>
  <div
    className="overlay"
  />
  <video
    autoPlay="autoplay"
    loop="loop"
    muted="muted"
    playsInline="playsinline"
  >
    <source
      src="/videos/bgv.mp4"
      type="video/mp4"
    />
  </video>
  <div
    className="container h-100"
  >
    <div
      className="d-flex h-100 text-center align-items-center"
    >
      <div
        className="textArea w-100 text-white"
      >
        <h1
          className="display-3"
        >
          Shangye
          <strong
            className="orangeText"
          >
            .space
          </strong>
        </h1>
        <p
          className="lead mb-0"
        >
          ALWAYS BE
          <br />
          YOURSELF
        </p>
      </div>
    </div>
  </div>
</section>
`;
